addons/sale/models/product_template.py

```diff
@@ -254,6 +257,10 @@ def _get_combination_info(self, combination=False, product_id=False, add_qty=1,
                 list_price, pricelist.currency_id, product_template._get_current_company(pricelist=pricelist),
                 fields.Date.today()
             )
+            price_extra = product_template.currency_id._convert(
+                price_extra, pricelist.currency_id, product_template._get_current_company(pricelist=pricelist),
+                fields.Date.today()
+            )
```

Comment:
Shouldn't the currency conversion be outside of the else part to convert the product price_extra and the template price_extra ?
Also, you should probably avoid doing the currency conversion if the currencies are the same, to avoid unwanted rounding changes :thinking:

Reply:
:+1:

Reply:
Forget what I said, it's already in a separate if checking the currency diff :facepalm: